core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1193,7 +1193,15 @@ class DAGScheduler(
             }
 
             if (runningStages.contains(shuffleStage) && shuffleStage.pendingPartitions.isEmpty) {
-              markStageAsFinished(shuffleStage)
+              val activeTaskSetManagerExist =
```

Comment:
And since it is being used as `!activeTaskSetManagerExists`, you could reverse the sense, avoid needing the `!`, and call it something like `noActiveTaskSetManager`.